### PR TITLE
qt: measure romfs dump completion by bytes read

### PR DESCRIPTION
**Pull request by liamwhite**
This improves the responsiveness of the dialog when dumping for games like SSBU which only have a single file in their romfs.

### CODE REVIEW

File: src/yuzu/main.cpp [hunk 1]
```diff
@@ -2275,40 +2275,61 @@ void GMainWindow::OnTransferableShaderCacheOpenFile(u64 program_id) {
...
+                dest_file->Resize(0);
+                return false;
+            }
+
+            using namespace std::literals::chrono_literals;
+            const auto new_timestamp = std::chrono::steady_clock::now();
+
+            if ((new_timestamp - last_timestamp) > 33ms) {
+                last_timestamp = new_timestamp;
+                dialog.setValue(std::min(read_size, total_size) * 100 / total_size);
```
Comment (FearlessTobi) [inline src/yuzu/main.cpp:2309]: in what case is it possible for read_size > total_size?

File: src/yuzu/main.cpp [hunk 2]
```diff
@@ -2667,12 +2687,15 @@ void GMainWindow::OnGameListDumpRomFS(u64 program_id, const std::string& game_pa
...
 
-    QProgressDialog progress(tr("Extracting RomFS..."), tr("Cancel"), 0,
-                             static_cast<s32>(entry_size), this);
+    QProgressDialog progress(tr("Extracting RomFS..."), tr("Cancel"), 0, 100, this);
     progress.setWindowModality(Qt::WindowModal);
     progress.setMinimumDuration(100);
+    progress.setAutoClose(false);
+    progress.setAutoReset(false);
+
+    size_t read_size = 0;
```
Comment (FearlessTobi) [inline src/yuzu/main.cpp:2696]: seems like this is unused?

Comment (liamwhite) [reply src/yuzu/main.cpp:2696]: it's used in the line below

File: src/yuzu/main.cpp (hunk 1 above)
Comment (liamwhite) [reply src/yuzu/main.cpp:2309]: It should never happen, it's just defensive programming.

File: src/yuzu/main.cpp (hunk 2 above)
Comment (FearlessTobi) [reply src/yuzu/main.cpp:2696]: ah, i didn't see that this is designed as a recursive function

**Review by FearlessTobi — APPROVED** [review]

**Pull request merged**